### ISSUE TITLE
Installing kontena-agent on Ubuntu 14.04 breaks Docker 1.7

### ISSUE THREAD

**Issue by ilkka** [kontena/kontena#83]
tl;dr: docker won't start due to malformed (?) DOCKER_OPTS after kontena-agent install

I have a VPS where `lsb_release -a` reports "Ubuntu 14.04.2 LTS" and docker-engine is at version 1.7.1-0~trusty. Adding the Kontena archive and installing kontena-agent modifies my `/etc/default/docker` such that DOCKER_OPTS is

    DOCKER_OPTS="--bridge=weave --fixed-cidr='10.81.null.0/24' --insecure-registry='10.81.0.0/16'"

and the Docker service fails to start. The "null" bit is suspicious, no?

    $ sudo docker -d --bridge=weave --fixed-cidr='10.81.null.0/24' --insecure-registry='10.81.0.0/16'
    INFO[0000] Listening for HTTP on unix (/var/run/docker.sock)
    INFO[0000] [graphdriver] using prior storage driver "aufs"
    WARN[0000] Running modprobe bridge nf_nat failed with message: , error: exit status 1
    FATA[0000] Error starting daemon: Error initializing network controller: invalid CIDR address: 10.81.null.0/24

Okay, let's take a guess at a more reasonable value:

```
$ sudo docker -d --bridge=weave --fixed-cidr='10.81.0.0/24' --insecure-registry='10.81.0.0/16'
INFO[0000] Listening for HTTP on unix (/var/run/docker.sock)
INFO[0000] [graphdriver] using prior storage driver "aufs"
WARN[0000] Running modprobe bridge nf_nat failed with message: , error: exit status 1
FATA[0000] Error starting daemon: Error initializing network controller: Error creating default "bridge" network: bridge device with non default name weave must be created manually
```

Shucks :( Maybe the modprobe warning is the problem?

```
$ lsmod|egrep '^(bridge|nf_nat)\b'
bridge                110833  0
nf_nat                 21841  5 nf_nat_ftp,ipt_MASQUERADE,nf_nat_ipv4,xt_nat,iptable_nat
```

wat :(

**Comment by ilkka** [comment]
Also FWIW the machine already seems to have a bridge IF called "weave".

**Comment by jakolehm** [comment]
It seems that kontena agent didn't get proper subnet from the master server. You should check that agent node can reach master and reinstall packages. Also trying to start docker directly won't work because it does not trigger upstart scripts.. so try to use `sudo start/stop docker` if you want to hack those manually.

**Comment by jakolehm** [comment]
Error seems to happen when there is no `eth1` interface present. Workaround is to set `WEAVE_PEER_INTERFACE=eth0` to /etc/default/kontena-weave file and re-install kontena-weave package.
